### PR TITLE
F/SRV-390 - Add Change Control CI Notification Support

### PR DESCRIPTION
**Pull request by afaq-sf**
### New Behavior
`change-management-event` job is added to notify EventHub and an Azure Function for production deployments.

This job:
* gathers information from the Pull Request (via danger.js) 
* sends payload to EventHub
* sends payload to Azure Function
* posts the same payload on the PR itself (via danger.js)

New keys required in `microservices-okta` context: 
* `PROD_CHANGE_ORDER_FUNC_URL`
* `PROD_CHANGE_ORDER_EVENTHUB_CONNECTION_STRING`
* `PROD_CHANGE_ORDER_EVENTHUB_NAME`

### Usage
Current MS circleci configs should be updated to add `change-management-event` job before the `deploy` job and filtered to `prod` branch only.

### Jira
https://mgmdigitalventures.atlassian.net/browse/SRV-390

### CODE REVIEW

**Review by misiek-sf — DISMISSED** [review]
one thought; if the 'danger' will fail PR then it will need to re-run it, can i see how this will be achieved? 
~also, some link to the "example" would be nice~ https://github.com/MGMResorts/parking/pull/101

**Comment by afaq-sf** [comment]
@misiek-sf Glad you found that.
I'll also be doing another test run after the `microservices-okta` context in CircleCI is updated with real-world values.

Will share the results of that here before we merge.

**Comment by afaq-sf** [comment]
Status update: 
Testing revealed that Azure is not responding when using the real-world prod EventHub configuration. The call to eventHub hangs forever until CircleCI kills the build.

Marking it as WIP accordingly until debugging / fix is complete.

**Comment by afaq-sf** [comment]
@moperacz @tomasz-sf @patrick-sf Please re-review this. It is tested to be functional.

You can see success and failure runtime here: https://github.com/MGMResorts/parking/pull/101

**Review by patrick-sf — DISMISSED** [review]

File: dangerfile.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ *
+ * @description
+ * This dangerfile validates production PRs before the deployment step
+ * to assure all change order requirements are met.
+ * The data captured here is then forwarded to Azure.
+ *
+ * See the orb commands for reference:
+ */
+const danger = require('danger')
+const { checkAssignees } = require('danger-plugin-complete-pr')
```
Comment (codin-sf) [inline dangerfile.js:12]: This is quite funny. I was searching around for this file and was confused. Then realized this is a "thing" https://www.npmjs.com/package/danger-plugin-complete-pr

**Review by codin-sf — DISMISSED** [review]
Looks great - I really hope this reduces our overhead on creating the CA tickets.

File: dangerfile.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+prodChangeEvent.requestor = danger.github.pr.assignee.login
+
+// Capture Deployment Summary
+if(hasDeploymentSummary()) {
+
+  prodChangeEvent.deploymentSummaryText = danger.github.pr.body.split(deploymentSummarySectionTitle)[1]
+
+  sendEventToChangeMgmtSystem(prodChangeEvent)
+    .then(r =>
+      message(`All production-dangerfile checks passed. A production change event has been sent to MGM Change Management system. ${githubFormatJson(prodChangeEvent)}`)
```
Comment (tomasz-sf) [inline dangerfile.js:70]: I don't see those `message` and `fail` functions imported anywhere. It should be imported from `danger` library, right? How does it work?

**Review by misiek-sf — DISMISSED** [review]

File: dangerfile.js (hunk 2 above)
Comment (afaq-sf) [reply dangerfile.js:70]: Dangerfile is evaluated inside of danger's own runtime with custom DSL.
The entire danger library and its globals are automatically injected and don't need to be imported.

https://danger.systems/js/tutorials/transpilation.html#the-quot-danger-quot-module

If your point is about styling/clarity, I can add in the requires.

**Pull request merged**